students/ScottL/session09/cmi_main.py

```diff
@@ -0,0 +1,155 @@
+    REPORT_HEADER_2 = f"\n{'-' * len(REPORT_HEADER_1)}"
+    DONOR_LIST_HEADER = f"\nDonor Name:\n{'-' * 20}"
+
+    def display_user_menu(self):
+        """Display the contents of the menu options for the user."""
+        print(self.MENU_HEADER_1, self.MENU_HEADER_2)
+        for item in self.MENU_SELECTIONS:
+            print(f"{item}: {self.MENU_SELECTIONS[item]}")
+
+    @staticmethod
+    def get_donor_name(first_last):
+        """
+        :param first_last: string to indicate which name (first or last) to supply
+        :return: string response to prompt
+        """
+        prompt = f"Please enter the {first_last} name of the donor, " \
+                 "'List' to see a list of existing donors, " \
+                 "or 'Quit' to return to the Main Menu: "
+        return input(prompt)
+
+    def display_donor_list(self):
+        """Display all the stored donor names for the user."""
+        if md.DonorCollection().name_list():
+            print(self.DONOR_LIST_HEADER)
+            for name in md.DonorCollection().name_list():
+                print(name)
+        else:
+            print("\n\tThe name list is empty.")
+
+    def validate_name(self):
+        """
+        Process the expected user input
+        :return: 2-element tuple containing the first and last names
+        """
+        first = self.get_donor_name("FIRST")
+        if first.strip().lower() not in ("quit", ""):
+            if first.strip().lower() == "list":
+                self.display_donor_list()
+            else:
+                last = self.get_donor_name("LAST")
+                if last.strip().lower() == "list":
+                    self.display_donor_list()
+                else:
+                    return first, last
+
+    @staticmethod
+    def get_donation_amount():
+        """
+        Receive and process input from user on the donation amount.
+        :return: float-type number of donation amount
+        """
+        prompt = "Enter the amount of the donation: "
+        amount = input(prompt)
+        value_error_message = "\nThe entry for 'amount' count was not a valid number. " \
+                              "Please try again."
+        if amount.strip().lower() not in ("quit", ""):
+            try:
+                amount = "".join(amount.split(","))
+                amount = float(amount.strip("$"))
+            except ValueError:
+                print(value_error_message)
+            else:
+                if amount < 0:
+                    print("\nUser attempted to enter a negative number; "
+                          "please input a positive number for the donation.")
+                else:
+                    return amount
+
+    def new_donation(self):
+        """Organize process flow for calling and validating user input for new donations"""
+        try:
+            first, last = self.validate_name()
+        except TypeError:
+            self.main_menu()
```

Comment:
you want to return, not call the main_menu as you'll get some weird exiting logic and you try to return back up the stack.